automerge-backend/src/change.rs

```diff
@@ -155,14 +178,39 @@ fn encode_chunk(
     }
 }
 
+#[derive(PartialEq, Debug, Clone)]
+enum ChangeBytes {
+    Compressed {
+        compressed: Vec<u8>,
+        uncompressed: Vec<u8>,
```

Comment:
Good point. Written as is the code makes a tradeoff of increased memory usage for the sake of faster encoding. This is useful in FFI where we very often have to encode the change before sending it across the FFI boundary as a bunch of bytes. An alternative way to implement this would be to have `ChangeBytes::raw_bytes` compress the content on demand. I've elected not to do that as it would imply compressing on every call which might increase latency on the FFI boundary. 

I think really we'll need to see how this performs and decide which makes most sense based on benchmarks.

Reply:
Ah, makes sense for optimising the FFI case then :)